security/identity-and-access-management/oracle-identity-governance/webservice-samples/extend-access-ws/files/src/com/oracle/samplews/ExtendAccessWSv1.java

```diff
@@ -0,0 +1,348 @@
+
+	/**
+	 * Connect to the remote/local OIM node using an OIMClient object
+	 * 
+	 * @throws IOException
+	 * @throws LoginException
+	 */
+	private void initOIMClient() throws IOException, LoginException {
+		logger.log(Level.FINE, "Initializing OIM Client...");
+
+		Properties prop = new Properties();
+		InputStream configurationsFileInput = null;
+
+		String wlDomainDirectory = System.getProperty("domain.home");
+		configurationsFileInput = new FileInputStream(
+				java.nio.file.Paths.get(wlDomainDirectory, "config", "extendaccessws.properties").toString());
+		prop.load(configurationsFileInput);
+
+		try {
+			configurationsFileInput.close();
+		} catch (IOException e) {
+			logger.log(Level.WARNING, "Unable to close properties file.");
+		}
+
+		String wsOIGNodeURL = prop.getProperty("wsOIGNodeURL");
+		logger.log(Level.FINE, "wsOIGNodeURL: {0}", wsOIGNodeURL);
+		String wsOIGUser = prop.getProperty("wsOIGUser");
+		logger.log(Level.FINE, "wsOIGUser: {0}", wsOIGUser);
+		String wsOIGPassword = prop.getProperty("wsOIGPassword");
+
+		Hashtable<String, String> env = new Hashtable<>();
+		env.put(OIMClient.JAVA_NAMING_FACTORY_INITIAL, "weblogic.jndi.WLInitialContextFactory");
+		env.put(OIMClient.JAVA_NAMING_PROVIDER_URL, wsOIGNodeURL);
+		oimClient = new OIMClient(env);
+		oimClient.login(wsOIGUser, wsOIGPassword.toCharArray());
+
+		logger.log(Level.FINE, "OIM Client initialization complete.");
+	}
+
+	/**
+	 * Initiate a DB connection. Note that the web service needs to be deployed on a
+	 * WebLogic node which also has access to the jdbc/operationsDB datasource, also
+	 * known as oimOperationsDB. This datasource is typically deployed only on the
+	 * OIM and SOA nodes of a WebLogic cluster.
+	 * 
+	 * @return
+	 * @throws IOException
+	 * @throws SQLException
+	 * @throws NamingException
+	 */
+	private Connection getDBConnection() throws IOException, SQLException, NamingException {
+		logger.log(Level.FINE, "Initializing DB Datasource connection...");
+
+		Properties prop = new Properties();
+		InputStream configurationsFileInput = null;
+
+		String wlDomainDirectory = System.getProperty("domain.home");
+		configurationsFileInput = new FileInputStream(
+				java.nio.file.Paths.get(wlDomainDirectory, "config", "extendaccessws.properties").toString());
+		prop.load(configurationsFileInput);
+
+		try {
+			configurationsFileInput.close();
+		} catch (IOException e) {
+			logger.log(Level.WARNING, "Unable to close properties file.");
+		}
+
+		String wsExtendAccessNodeURL = prop.getProperty("wsExtendAccessNodeURL");
+		logger.log(Level.FINE, "wsExtendAccessNodeURL: {0}", wsExtendAccessNodeURL);
+
+		Hashtable<Object, String> env = new Hashtable<>();
+		env.put(OIMClient.JAVA_NAMING_FACTORY_INITIAL, "weblogic.jndi.WLInitialContextFactory");
+		env.put(OIMClient.JAVA_NAMING_PROVIDER_URL, wsExtendAccessNodeURL);
+		Context ctx = new InitialContext(env);
+		DataSource ds = (javax.sql.DataSource) ctx.lookup("jdbc/operationsDB");
```

Comment:
Hidden imports in the code are not a good coding practice. Add this class to the import section.

Reply:
Fixed, as above.